### PR TITLE
remove pyjq requirement and make pip installable

### PR DESCRIPTION
**Pull request by ryandeivert**
@0xdabbad00 
resolves: #31, #32

### Reason for change(s)

Being dependent on the `pyjq` library makes the setup/usage of `cloudtracker` a bit clunky and not platform-independent. This is because `pyjq` itself has core dependencies that must compiled on the target platform, which requires extra tooling such as `autoconf`, `automake`, `libtool`, etc. (See issue #31)

The main reason for this change is to enable #32 - making `cloudtracker` installable via pip.

### Changes

Given that `pyjq` is only used a handful of times, and in a fairly simplistic fashion, I have migrated cloudtracker to use `jmespath`, which allows for similar functionality.

Adding `setup.py` and `setup.cfg` to enable installing via setuptools/pip.
- Updating README.md to include information on new installation flow.
- Adding ability to run unit tests with nose via `python setup.py test`
- Removing now unnecessary unit test shell script
- Removing needs for `Makefile` for supporting elasticsearch
  - The default install will _not_ include elasticsearch support
  - Support for elasticsearch will be included (on merge) by installing via:
    - `pip install git+https://github.com/duo-labs/cloudtracker.git#egg=cloudtracker[es6]` (or [es1])
  - Updating elasticsearch docs to include note on this.

Moving cli code to within package to enable proper packaging    
- Adding entry point for console script that will expose a `cloudtracker` executable to be used instead of using `python cloudtracker.py`
- Updating README.md to reflect changes and to note new executable.
- Updating some argparse logic to improve UX and reduce extra user-defined logic for validation/file reading.

Using the setuptools feature of `package_data` to include `aws_api_list.txt` and `cloudtrail_supported_actions.txt` files
- As a best practice, dependent data should be packaged with the installed package.
- The text data files needed now live in `cloudtracker/data` and can be accessed using the `pkg_resources` package.

### Testing

Updating some invalid unit tests.
Installed locally via `pip install git+file:///path/to/repos/cloudtracker@ryandeivert-make-pip-installable`

and with elasticsearch support via:

`pip install -e git+file:///Users/ryandeivert/repos/cloudtracker@ryandeivert-make-pip-installable[es1]`

Tested commands with account data with success.

### CODE REVIEW

File: cloudtracker/__init__.py [hunk 1]
```diff
@@ -228,12 +229,15 @@ def get_user_allowed_actions(aws_api_list, user_iam, account_iam):
...
+        if group_iam is None:
+            continue
         # Get privileges from managed policies attached to the group
         for managed_policy in group_iam['AttachedManagedPolicies']:
-            policy_filter = '.Policies[] | select(.Arn == "{}") | ' \
-                            '.PolicyVersionList[] | select(.IsDefaultVersion == true) | .Document'
-            policy = pyjq.one(policy_filter.format(managed_policy['PolicyArn']), account_iam)
+            policy_filter = 'Policies[?Arn == `{}`].PolicyVersionList[?IsDefaultVersion == true] | [0][0].Document'
+            policy = jmespath.search(policy_filter.format(managed_policy['PolicyArn']), account_iam)
+            if police is None:
```
Comment (0xdabbad00) [inline cloudtracker/__init__.py]: Looks like a typo here where `police` should be `policy`

Comment (ryandeivert) [reply cloudtracker/__init__.py]: definitely a typo, sorry about that 🤣

**Comment by ryandeivert** [comment]
@0xdabbad00 just pushed a commit to fix the typo

**Comment by ryandeivert** [comment]
thanks scott! I tagged this issue wrong but this should also resolve #32 :)

**Comment by 0xdabbad00** [comment]
I'm leaving that issue open until I actually make this pip installable (ie. `pip install cloudtracker` without needing a URL).  I'm digging into that more now.

**Pull request merged**